server/models/services/GithubWorkflowService.js

```diff
@@ -0,0 +1,157 @@
+// > future.
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+const calculateExpiresAt = () => Math.round(Date.now() / 1000) + 9 * ONE_MINUTE;
+
+const fetchInstallationAccessToken = async (jsonWebToken, installationID) => {
+    const response = await fetch(
+        `https://api.github.com/app/installations/${installationID}/access_tokens`,
+        {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.github+json',
+                Authorization: `Bearer ${jsonWebToken}`,
+                'X-GitHub-Api-Version': '2022-11-28'
+            }
+        }
+    );
+
+    if (!response.ok) {
+        const message = await (() => {
+            try {
+                return response.text();
+            } catch (_) {
+                return 'Unable to retrieve installation access token';
+            }
+        })();
+
+        throw new Error(message);
+    }
+
+    return (await response.json()).token;
+};
+
+const createGithubWorkflow = async ({
+    job,
+    directory = 'alert',
+    gitSha = 'master'
+}) => {
+    const payload = {
+        iat: calculateIssuedAt(),
+        exp: calculateExpiresAt(),
+        iss: GITHUB_APP_ID
+    };
+    const jsonWebToken = await createJWT(payload, privateKey, ALGORITHM);
+    const accessToken = await fetchInstallationAccessToken(
+        jsonWebToken,
+        GITHUB_APP_INSTALLATION_ID
+    );
+    const inputs = {
+        nonce: job.id,
+        callback_url: `https://${process.env.AUTOMATION_CALLBACK_FQDN}/api/jobs/${job.id}/result`,
+        status_url: `https://${process.env.AUTOMATION_CALLBACK_FQDN}/api/jobs/${job.id}/update`,
+        // TODO: This should probably just be configured instead of us passing it to the scheduler every time
+        callback_header: `x-automation-secret:${process.env.AUTOMATION_SCHEDULER_SECRET}`,
+        work_dir: `tests/${directory}`,
+        test_pattern: 'reference/**,test-*-nvda.*',
+        aria_at_ref: gitSha
+    };
+    const response = await fetch(
+        `https://api.github.com/repos/${WORKFLOW_REPO}/actions/workflows/${WORKFLOW_FILE_NAME}/dispatches`,
+        {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.github+json',
+                Authorization: `Bearer ${accessToken}`,
+                'X-GitHub-Api-Version': '2022-11-28'
+            },
+            body: JSON.stringify({
+                ref: 'main',
+                inputs
+            })
+        }
+    );
+
+    if (!response.ok) {
+        const message = await (() => {
+            try {
+                return response.text();
+            } catch (_) {
+                return 'Unable to initiate workflow';
+            }
+        })();
+
+        throw new Error(message);
+    }
+
+    // TODO: Listen for GitHub Webhook event describing the new workflow, and
+    // return its ID.
+    return true;
```

Comment:
I believe this is no longer the plan. If so, we can remove the comment and the return value.

Reply:
there isn't a "value" here; basically success or throw; so left `return true`